sw/common/pwm.h

```diff
@@ -0,0 +1,17 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef PWM_H__
+#define PWM_H__
+
+#include <stdint.h>
+
+typedef uint32_t* pwm_t;
+
+#define PWM_FROM_ADDR_AND_INDEX(addr, index)((pwm_t) (addr + sizeof(pwm_t)*index))
+#define NUM_PWM_MODULES 12
```

Comment:
This should live in `demo_system.h`. This file is for the generic PWM 'driver' where `demo_system.h` is used for the specifics of the demo system (so someone wanting to make changes or reuse the IP in a different top-level has a clean separation between IP things and demo system things).

Reply:
Good, point. I've moved this now.